src/payment/strategies/digitalriver/digitalriver-payment-strategy.ts

```diff
@@ -84,14 +86,25 @@ export default class DigitalRiverPaymentStrategy implements PaymentStrategy {
 
     async execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
         const { payment, ...order } = orderRequest;
+        const { useStoreCredit } = order;
 
-        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+        await this._store.dispatch(this._orderActionCreator.submitOrder( { useStoreCredit }, options));
```

Comment:
passing `{useStoreCredit}` in the order creation is deprecated in favour of using `checkoutService.applyStoreCredit()`.

We have left it on existing implementations but discouraged in new implementations.

Reply:
Hey @capsula4 thanks, I added that based on [Zip, Bolt] strategy, also covered testing, It's ready for another review ♻️